diff -Nuar glm.orig/glm/detail/setup.hpp glm/glm/detail/setup.hpp
--- glm.orig/glm/detail/setup.hpp	2015-02-15 16:22:30.000000000 +0100
+++ glm/glm/detail/setup.hpp	2015-03-15 22:12:24.041950002 +0100
@@ -237,7 +237,8 @@
 #		elif __clang_major__ >= 6
 #			define GLM_COMPILER GLM_COMPILER_APPLE_CLANG60
 #		endif
-#	else
+#       endif
+#       if !defined(GLM_COMPILER)
 #		if __clang_major__ == 3 && __clang_minor__ == 0
 #			define GLM_COMPILER GLM_COMPILER_LLVM30
 #		elif __clang_major__ == 3 && __clang_minor__ == 1
